rfcs/0013-data-plane-fanout.md

```diff
@@ -0,0 +1,86 @@
+Validators know to only forward blobs that came from the leader by checking the signatures against the current leader.   
+
+**Cluster_info -> retransmit** = Used by TVUs to retransmit. Currently Layer-1 sends this to all tvu peers, leader is automatically excluded.
+
+**Cluster_info -> broadcast** = Used by leader to broadcast to layer-1 nodes. 
+
+**BroadcastStage -> run** = Used by leader (TPU) to broadcast to all validators. Currently all TVU Peers are considered layer-1. But blobs are transmitted sort of round robin. See Cluster_info->Broadcast. 
+
+
+## Proposed Design
+
+The new design organizes the network by stake and divides it into a collection of nodes, called `neighborhoods`. 
+The leader broadcasts its blobs to the layer-1 (neighborhood 0) nodes exactly like it does without this mechanism. The main difference being the number of nodes in layer-1 is capped via the configurable `DATA_PLANE_FANOUT`. If the fanout is smaller than the nodes in the network then the mechanism will add layers below layer-1. Subsequent layers (beyond layer-1) follow the following constraints to determine layer-capacity. 
+Each neighborhood has `NEIGHBORHOOD_SIZE` nodes and `fanout/2` neighborhoods are allowed per layer. 
+
+Nodes in a layer will broadcast their blobs to exactly 1 node in each next-layer neighborhood and each node in a neighborhood will perform retransmits amongst its neighbors (just like layer-1 does with its layer-1 peers).
+This means any node has to only send its data to its neighbors and each neighborhood in the layer below instead of every single tvu peer it has.  
+The retransmit mechanism also supports a second, `grow`,  mode of operation that squares the number of neighborhoods allowed per layer which dramatically reduces the number of layers needed to support a large network but can also have a negative impact on the network pressure each node in the lower layers has to deal with.  
+
+Below is an example of a two layer network. Note - this example doesn't describe the same `fanout/2` limit for lower layer neighborhoods.
+               
+                                              +--------------+
+                                              |              |
+                                 +------------+    Leader    +------------+
+                                 |            |              |            |
+                                 |            +--------------+            |
+                                 v                                        v
+                        +--------+--------+                      +--------+--------+
+                        |                 +--------------------->+                 |
+      +-----------------+   Validator 1   |                      |   Validator 2   +-------------+
+      |                 |                 +<---------------------+                 |             |
+      |                 +------+-+-+------+                      +---+-+-+---------+             |
```

Comment:
Why all these outbound arrows and then no arrows between peers in the next level? I don't understand how there'd be sufficient bandwidth to send all those copies at peak load. Also, with this visualization, I don't see how the algorithm between level 0 and level 1 is the same as between level 1 and level 2.

Reply:
I wonder if it would be sufficient to have just 2 smaller diagrams:

1. A 2-level diagram where the top node is Neighborhood 0 and layer 1 is Neighborhood 1 and Neighborhood 
2. A diagram showing how the blob enters a neighborhood, is passed between peers, and retransmitted to the next neighborhood.

I also like Rob's idea of showing the path of just one blob. I think we need some way to visualize that blob 0 starts its journey in one tick and that blob 1 starts in another.  Perhaps a traditional pipeline diagram would be helpful in showing how blob 0 and blob 1 can coexist in different parts of the data plane without creating additional network pressure.